pkg/api/steve/supportconfigs/generator_test.go

```diff
@@ -0,0 +1,182 @@
+			generateAdapterError:    true,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusInternalServerError,
+		},
+		{
+			name:                    "test default unauthorized due to error while performing authorization",
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: true,
+			authorized:              false,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusForbidden,
+		},
+		{
+			name:                    "test user requests PAYG, authorized to get PAYG, PAYG not installed (501)",
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: true,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusNotImplemented,
+		},
+		{
+			name:                    "test user requests PAYG, not authorized to get PAYG, auth denied (403)",
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: true,
+			generateAuthorizedError: false,
+			authorized:              false,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusForbidden,
+		},
+		{
+			name:                    "test user requests BYOL, authorized to get BYOL, BYOL not installed (501)",
+			usePAYG:                 false,
+			generateAdapterError:    false,
+			generateAdapterNotFound: true,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusNotImplemented,
+		},
+		{
+			name:                    "test user requests BYOL, not authorized to get BYOL, auth denied (403)",
+			usePAYG:                 false,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: false,
+			authorized:              false,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusForbidden,
+		},
+		{
+			name:                    "test user requests BYOL, authorized to get BYOL, we return output (200)",
+			usePAYG:                 false,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusOK,
+		},
+		{
+			name:                    "test user requests PAYG, autorized to get PAYG, we return output (200)",
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusOK,
+		},
+	}
+	for _, scenario := range scenarios {
+		test := scenario
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
You could also run each subtest in parallel:
```suggestion
		t.Run(test.name, func(t *testing.T) {
		    t.Parallel()
```